clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/SecurityGroupIngressConverter.groovy

```diff
@@ -130,4 +127,48 @@ class SecurityGroupIngressConverter {
       }
     }.flatten().unique()
   }
+
+  static Map<String, List> computeLatestIpRules(List<IpPermission> newList, List<IpPermission> existingRules) {
+    List<IpPermission> tobeAdded = new ArrayList<>()
+    List<IpPermission> tobeRemoved = new ArrayList<>()
+    List<IpPermission> tobeUpdated = new ArrayList<>()
+    List<IpPermission> filteredList = newList.findAll { elements -> elements.userIdGroupPairs.size() == 0 }
+    List<IpPermission> filteredExisistingRuleList = existingRules.findAll { elements -> elements.userIdGroupPairs.size() == 0 }
+    filteredList.forEach({ newListEntry ->
+      IpPermission match = findIpPermission(filteredExisistingRuleList, newListEntry)
+      if (match) {
+        if (newListEntry.ipv4Ranges.collect { it.description }.find() != null
+          || newListEntry.ipv6Ranges.collect { it.description }.find() != null) {
+          tobeUpdated.add(newListEntry) // matches old rule , needs an update for description
+          filteredExisistingRuleList.remove(newListEntry) // remove from future processing
+        }
+        filteredExisistingRuleList.remove(match) // remove from future processing
+      } else {
+        tobeAdded.add(newListEntry) //no match in old rule so must be added
+      }
+    })
+    tobeRemoved = filteredExisistingRuleList // rules that needs to be removed
+    Map<String, List> modificationsMap = ["tobeAdded": tobeAdded, "tobeRemoved": tobeRemoved, "tobeUpdated": tobeUpdated]
+    modificationsMap
+  }
+
+  static IpPermission findIpPermission(List<IpPermission> existingList, IpPermission ipPermission) {
+    existingList.find { it ->
```

Comment:
why not rely on the [`IpPermission.equals`](https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-ec2/src/main/java/com/amazonaws/services/ec2/model/IpPermission.java#L697) method I think it does exactly what you want (but please double check)

Reply:
It does not exactly do what is required, i will leave this logic in , unless you feel we need to update it